internal/cache/controller_obj_cache.go

```diff
@@ -2597,6 +2605,19 @@ func (c *AviObjCache) AviObjVSCachePopulate(client *clients.AviClient, cloud str
 						}
 					}
 				}
+				if vs["pool_group_ref"] != nil {
```

Comment:
Is this something we’re adding to avoid it becoming stale?

Reply:
yes. Here we are going through each avi controller object (created by that AKO) and check if that is associated with any VS or not. Then we are deleting those objects who are not associated with VS